mt-kahypar/partition/initial_partitioning/recursive_bisection_initial_partitioner.cpp

```diff
@@ -332,16 +332,40 @@ namespace mt_kahypar {
       bisection_context.type = kahypar::ContextType::initial_partitioning;
 
       // Setup Part Weights
-      HypernodeWeight total_weight = hypergraph.totalWeight();
-      if ( context.initial_partitioning.use_adaptive_epsilon ) {
-        bisection_context.partition.epsilon = _original_hypergraph_info.computeAdaptiveEpsilon(
-                total_weight, context.partition.k);
+      const HypernodeWeight total_weight = hypergraph.totalWeight();
+      const PartitionID k = context.partition.k;
+      const PartitionID k0 = k / 2 + (k % 2 != 0 ? 1 : 0);
+      const PartitionID k1 = k / 2;
+      ASSERT(k0 + k1 == context.partition.k);
+      if ( context.partition.use_individual_part_weights ) {
+        const HypernodeWeight max_part_weights_sum = std::accumulate(context.partition.max_part_weights.cbegin(),
+                                                                    context.partition.max_part_weights.cend(), 0);
+        const double weight_fraction = total_weight / static_cast<double>(max_part_weights_sum);
+        ASSERT(weight_fraction <= 1.0);
+        bisection_context.partition.perfect_balance_part_weights.clear();
+        bisection_context.partition.max_part_weights.clear();
+        HypernodeWeight perfect_weight_p0 = 0;
+        for ( PartitionID i = 0; i < k0; ++i ) {
+          perfect_weight_p0 += ceil(weight_fraction * context.partition.max_part_weights[i]);
+        }
+        HypernodeWeight perfect_weight_p1 = 0;
+        for ( PartitionID i = k0; i < k; ++i ) {
+          perfect_weight_p1 += ceil(weight_fraction * context.partition.max_part_weights[i]);
+        }
+        const double base = max_part_weights_sum / static_cast<double>(perfect_weight_p0 + perfect_weight_p1);
+        bisection_context.partition.epsilon = total_weight == 0 ? 0 : std::min(0.99, std::max(std::pow(base, 1.0 /
```

Comment:
the adaptive epsilon computation here does not consider the hypergraph weight of the current subhypergraph and the original hypergraph. Is this correct?

Reply:
maybe add a comment ...

Reply:
Well, sort of. I have added a lengthier comment now that hopefully clears it up.